test/testers.jl

```diff
@@ -42,62 +42,42 @@ primalapprox(x) = x
         end
     end
 
-    @testset "Inplace accumumulation: first on Array" begin
+    @testset "Inplace accumulation: identity on Array" begin
         @testset "Correct definitions" begin
-            function ChainRulesCore.frule((_, ẋ), ::typeof(first), x::Array)
-                ẏ = InplaceableThunk(
-                    @thunk(first(ẋ)),
-                    ȧ -> ȧ + first(ẋ),  # This won't actually happen inplace
```

Comment:
I guess an outdated comment, but that is only true for the current input, right? (i.e. `rand(4)`).

My understanding it that `first` would not make a copy of an array, because of this:
```
julia> a = rand(3)
3-element Array{Float64,1}:
 0.36709403077151537
 0.49246683838177385
 0.8083301928782081 

julia> A = [a, rand(3)]
2-element Array{Array{Float64,1},1}:
 [0.36709403077151537, 0.49246683838177385, 0.8083301928782081]
 [0.822528854921962, 0.6952904499081514, 0.43481292252062986]  

julia> first(A) === a
true
```
But maybe this isn't testing what I think it is testing.

Reply:
the reason it isn't acxtually happening inplace is because because it doesn't mutate the content of `ȧ`